baseframe/utils.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+import six
+from six.moves.urllib.parse import quote as urlquote
+
+from mxsniff import mxsniff, MXLookupException
+
+from . import asset_cache
+
+__all__ = ['is_public_email_domain']
+
+
+def is_public_email_domain(email_or_domain, default=None, timeout=30):
+    """
+    Checks if the given domain (or domain of given email) is known to offer public email accounts
+
+    :param email_or_domain: email address or domain name to check
+    :param default: default value to return in case domain lookup fails
+    :param timeout: Lookup timeout (will raise an exception if timeout happens before the check is completed and no default is provided)
```

Comment:
The parenthesised bit can be moved into the main documentation. Also add `:raises MXLookupException: If a DNS lookup error happens and no default is specified`